rules/sql-server-memory-pressure/rule.md

```diff
@@ -0,0 +1,92 @@
+---
+type: rule
+archivedreason: 
+title: What to do about SQL Server Memory Pressure?
+guid: f7e8a48d-375e-4c23-9998-9cd9b8c7980c
+uri: sql-performance-memory-pressure
+created: 2023-12-27T07:26:54.0000000Z
+authors:
+  - title: Bryden Oliver
+    url: https://ssw.com.au/people/bryden-oliver
+related: 
+  - identify-sql-performance-sql-server
+  - identify-sql-performance-azure
+  - sql-server-memory-pressure
+  - sql-server-io-pressure
+  - sql-server-cpu-pressure
+---
+So you've identified that your SQL Server is under memory pressure. What can you do about it?
+
+<!--endintro-->
+
+If SQL Server is the primary consumer, then read up on reducing SQL Server memory usage, or whether more memory is appropriate for your workload.
+
+## Identify non SQL engine related usage
+
+Run
+
+```sql
+select * from sys.dm_os_memory_clerks where type='MEMORYCLERK_HOST'
+```
+
+This isolates a few of the SQL processes that aren't part of the SQL Server engine.
+Look for high memory usage for OLE DB providers (MSOLEDBSQL), SQL Native Client (SQLNCLI*) and so on.
+This may indicate using some non core features and you should evaluate whether these are necessary.
+
+## Identify SQL engine related usage
+
```

Comment:
**question**: Should there be an introductory paragraph (for the section) before the SQL text?

Reply:
Yes there should :) And there now is...